src/index.js

```diff
@@ -97,136 +140,117 @@ class UglifyJsPlugin {
 						}
 						const stream = uglify.OutputStream(output); // eslint-disable-line new-cap
 						ast.print(stream);
-						if(map) map = map + "";
-						const stringifiedStream = stream + "";
-						let outputSource = (map ?
-							new SourceMapSource(stringifiedStream, file, JSON.parse(map), input, inputSourceMap) :
-							new RawSource(stringifiedStream));
+						if(map) map += "";
+						const stringifiedStream = `${stream}`;
+						let outputSource = map ? new SourceMapSource(
+              stringifiedStream, file, JSON.parse(map), input, inputSourceMap // eslint-disable-line comma-dangle
+            ) : new RawSource(stringifiedStream);
 						if(extractedComments.length > 0) {
-							let commentsFile = options.extractComments.filename || file + ".LICENSE";
+							let commentsFile = options.extractComments.filename || `${file}.LICENSE`;
 							if(typeof commentsFile === "function") {
 								commentsFile = commentsFile(file);
 							}
 
-							// Write extracted comments to commentsFile
-							const commentsSource = new RawSource(extractedComments.join("\n\n") + "\n");
+              // Write extracted comments to commentsFile
+							const commentsSource = new RawSource(`${extractedComments.join("\n\n")}\n`);
 							if(commentsFile in compilation.assets) {
-								// commentsFile already exists, append new comments...
+                // commentsFile already exists, append new comments...
 								if(compilation.assets[commentsFile] instanceof ConcatSource) {
 									compilation.assets[commentsFile].add("\n");
 									compilation.assets[commentsFile].add(commentsSource);
 								} else {
 									compilation.assets[commentsFile] = new ConcatSource(
-										compilation.assets[commentsFile], "\n", commentsSource
-									);
+                    compilation.assets[commentsFile], "\n", commentsSource
+                  );
 								}
 							} else {
 								compilation.assets[commentsFile] = commentsSource;
 							}
 
-							// Add a banner to the original file
+              // Add a banner to the original file
 							if(options.extractComments.banner !== false) {
-								let banner = options.extractComments.banner || "For license information please see " + commentsFile;
+								let banner = options.extractComments.banner || `For license information please see ${commentsFile}`;
 								if(typeof banner === "function") {
 									banner = banner(commentsFile);
 								}
 								if(banner) {
 									outputSource = new ConcatSource(
-										"/*! " + banner + " */\n", outputSource
-									);
+                    `/*! ${banner} */\n`, outputSource
+                  );
 								}
 							}
 						}
-						asset.__UglifyJsPlugin = compilation.assets[file] = outputSource;
-						if(warnings.length > 0) {
-							compilation.warnings.push(new Error(file + " from UglifyJs\n" + warnings.join("\n")));
+						compilation.assets[file] = outputSource;
+						asset.__UglifyJsPlugin = outputSource;
```

Comment:
Looks like this should be a `WeakMap` instead.

But fixing this is outside the scope of this PR.